src/core.c/Distribution/Resource.rakumod

```diff
@@ -22,7 +22,7 @@ class Distribution::Resource {
     }
 
     # delegate appropriate IO::Path methods to the resource IO::Path object
-    multi method Str(::?CLASS:D: |c) {
+    multi method Str(::?CLASS:D: |c) is DEPRECATED('%?RESOURCES<key> directly instead') {
```

Comment:
A potential improvement might be showing the key being accessed. For example: `is DEPRECATED(sprintf '%?RESOURCES<%s> directly instead', $!key)`

Reply:
It looks like this didn't quite work as is. I'm not sure how to access the instances attribute while defining a trait.

Reply:
Thanks for trying my suggestion. After checking this out I think your original commit is better since it (the trait specifically) will show the liner numbers of all the callers, whereas the code I suggested shows the line number of the core rakudo code where `DEPRECATED(...)` is called. If you could `git reset --hard 721e9afc088d714d3ef242c1afd7cde645b0bbd3` or just add a commit going back to your original code I'll accept and merge.

Reply:
Done! With some formatting changes. Thanks for the explanation!